Deprecate torch.lu_solve

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73878
* #73877
* #74046
* __->__ #73806
* #73804
* #73803
* #72935

**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

lezcano committed:
Deprecate torch.lu_solve

**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

IvanYashchuk commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -1979,10 +1979,26 @@ TORCH_IMPL_FUNC(linalg_lu_solve_out)(const Tensor& LU,
+  TORCH_WARN_ONCE(
+    "torch.lu_solve is deprecated in favor of torch.linalg.lu_solve",
+    "and will be removed in a future PyTorch release.\n",
+    "Note that torch.linalg.solve has its arguments reversed.\n",
```
```suggestion
    "Note that torch.linalg.lu_solve has its arguments reversed.\n",
```

Lezcano replied:
Aaah, yes, the copy-paste

IvanYashchuk commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -1979,10 +1979,26 @@ TORCH_IMPL_FUNC(linalg_lu_solve_out)(const Tensor& LU,
+  TORCH_WARN_ONCE(
+    "torch.lu_solve is deprecated in favor of torch.linalg.lu_solve",
+    "and will be removed in a future PyTorch release.\n",
+    "Note that torch.linalg.solve has its arguments reversed.\n",
```
```suggestion
    "Note that torch.linalg.lu_solve has its arguments reversed.\n",
```

lezcano committed:
Update aten/src/ATen/native/BatchLinearAlgebra.cpp

Co-authored-by: Ivan Yashchuk <IvanYashchuk@users.noreply.github.com>

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu_solve"


**BC-breaking note**:

This PR deprecates `torch.lu_solve` in favor of `torch.linalg.lu_solve_factor`.
A upgrade guide is added to the documentation for `torch.lu_solve`.

Note this PR DOES NOT remove `torch.lu_solve`.

[ghstack-poisoned]

mruberry reviewed (approved):
Cool! Stamped!

Lezcano commented:
@pytorchmergebot merge this please

malfet commented:
@pytorchbot revert this, as dependent change of https://github.com/pytorch/pytorch/pull/72935 that breaks internal build

Lezcano commented:
Relanding in https://github.com/pytorch/pytorch/pull/77637
